rust/src/traits/samplers/psrn/tulap/InverseCDF.tex

```diff
@@ -0,0 +1,97 @@
+\documentclass{article}
+\input{../../../../lib.sty}
+
+\usepackage[
+        backend=biber,
+        style=authoryear-comp,
+        sorting=nyt,
+    ]{biblatex}
+
+\usepackage{thmtools}
+\usepackage{thm-restate}
+\newtheorem{thm}{Theorem}
+\newtheorem{defn}{Definition} 
+\newtheorem{prop}{Proposition}
+\newcommand{\mrm}[1]{\mathrm{#1}}
+
+
+\title{\texttt{trait impl InverseCDF for TulapRV}}
+\author{Yu-Ju Ku, Jordan Awan, Aishwarya Ramasethu, Michael Shoemate}
+\begin{document}
+
+\maketitle
+
+\contrib
+
+Proves soundness of \rustdoc{traits/samplers/psrn/tulap/struct}{TulapRV}.
+% in \asOfCommit{mod.rs}{f5bb719}.
+\texttt{edge} accepts parameter \texttt{self}, containing the state of the Tulap sampler and \texttt{R} specifying the rounding mode.
+\floatingPoint
+
+\begin{tcolorbox}
+    \begin{warning}[Code is not constant-time]
+        The implementation of \texttt{edge} uses procedures that are vulnerable to timing attacks. 
+    \end{warning}
+\end{tcolorbox}
+
+\subsection*{PR History}
+\begin{itemize}
+    \item \vettingPR{1126}
+\end{itemize}
+
+\section{Hoare Triple}
+
+\subsection*{Preconditions}
+\begin{itemize}
+    \item Variable \texttt{self} is of type \texttt{TulapRV}.
+    \item Generic \texttt{R} denotes the rounding mode, one of "up" or "down".
+\end{itemize}
+
+\subsection*{Pseudocode}
+
+\lstinputlisting[language=Python,firstline=2]{./pseudocode/InverseCDF.py}
+
+\subsection*{Postcondition}
+
+\texttt{edge} returns an upper or lower bound for the true Tulap sample,
+a distribution with CDF defined in \texttt{make\_tulap}.
+
+\section{Proof}
+
+\begin{proof} 
```

Comment:
Proof of what theorem? If it's the proposition, then maybe start the environment after stating the theorem.